src/zenml/steps/utils.py

```diff
@@ -318,6 +318,10 @@ def resolve_input_artifact(
         Returns:
             Return the output of `handle_input()` of selected materializer.
         """
+        # Make an exception for BaseArtifact, return artifact
```

Comment:
I think we should have tests for both having a (Base)Artifact as input as well as output. We might also want to check what happens if e.g. the previous step returns a ModelArtifact but your step defines the input as a DataArtifact.

Reply:
I exactly implemented this test without looking at this comment beforehand. Same thought! For now I have just raised a warning